tests/lib/rules/no-deprecated-router-transition-methods.js

```diff
@@ -0,0 +1,435 @@
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+const rule = require('../../../lib/rules/no-deprecated-router-transition-methods');
+const RuleTester = require('eslint').RuleTester;
+
+//------------------------------------------------------------------------------
+// Tests
+//------------------------------------------------------------------------------
+
+const ruleTester = new RuleTester({
+  parser: require.resolve('@babel/eslint-parser'),
+  parserOptions: {
+    ecmaVersion: 2020,
+    sourceType: 'module',
+    babelOptions: {
+      configFile: require.resolve('../../../.babelrc'),
+    },
+  },
+});
+
+ruleTester.run('no-deprecated-router-transition-methods', rule, {
```

Comment:
Since we're using the class stack, technically we should also be testing that the behavior is correct with nested classes. For example, that the behavior is correct inside the nested class, and after the nested class but still inside the outer class.

Reply:
Similar to the other PR, I'll add nested classes and sibling classes.